core/src/main/java/org/pentaho/di/connections/vfs/BaseVFSConnectionProvider.java

```diff
@@ -90,4 +101,65 @@ protected static boolean getBooleanValueOfVariable( VariableSpace space, String
   protected VariableSpace getSpace( ConnectionDetails connectionDetails ) {
     return connectionDetails.getSpace() == null ? Variables.getADefaultVariableSpace() : connectionDetails.getSpace();
   }
+
+  @Override
+  public boolean test( T connectionDetails, VFSConnectionTestOptions connectionTestOptions ) throws KettleException {
+    boolean valid = test( connectionDetails );
+    if ( !valid ) {
+      return false;
+    }
+
+    if ( !connectionDetails.isSupportsRootPath()  || connectionTestOptions.isIgnoreRootPath() ) {
+      return true;
+    }
+
+    String resolvedRootPath = getResolvedRootPath( connectionDetails );
+    if ( StringUtils.isEmpty( resolvedRootPath ) ) {
+      return !connectionDetails.isRootPathRequired();
+    }
+
+    String internalUrl = buildUrl( connectionDetails, resolvedRootPath );
+    FileObject fileObject = KettleVFS.getFileObject( internalUrl, new Variables(), getOpts( connectionDetails ) );
+
+    try {
+      return fileObject.exists();
```

Comment:
Must also validate that the file is a folder!

Reply:
@nawaz34-hitachivantara  here is a snippet that does the check folder type for Apache File Objects

https://github.com/pentaho/pentaho-kettle/blob/113d11c748a697a20fb4f23c4611655d4a2093fc/plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/vfs/VFSFileProvider.java#L293C41-L293C49

Reply:
Addressed the changes.

Reply:
Should check if file exists _before_ checking it is a folder!

Reply:
addressed the change